business/istio_certs.go

```diff
@@ -0,0 +1,143 @@
+package business
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/kubernetes"
+	"gopkg.in/yaml.v2"
+)
+
+type IstioCertsService struct {
+	k8s           kubernetes.ClientInterface
+	businessLayer *Layer
+}
+
+type IstioConfig struct {
+	Certificates []struct {
+		SecretName string   `yaml:"secretName"`
+		DNSNames   []string `yaml:"dnsNames"`
+	} `yaml:"certificates"`
+}
+
+type CertInfo struct {
+	SecretName string    `json:"secretName"`
+	DNSNames   []string  `json:"dnsNames"`
+	Issuer     string    `json:"issuer"`
+	Subject    string    `json:"subject"`
+	NotBefore  time.Time `json:"notBefore"`
+	NotAfter   time.Time `json:"notAfter"`
+	Error      error     `json:"error"`
+}
+
+func (ics *IstioCertsService) GetCertsInfo() ([]CertInfo, error) {
+	if certs, err := ics.getChironCertificates(); err == nil {
+		return certs, nil
+	} else if cert, err := ics.getCertificateFromSecret("cacerts"); err == nil {
+		return []CertInfo{cert}, nil
+	} else {
+		cert, err := ics.getCertificateFromSecret("istio-ca-secret")
+		if err != nil {
+			return nil, err
+		}
+		return []CertInfo{cert}, nil
+	}
+}
+
+func (ics *IstioCertsService) getCertificateFromSecret(secretName string) (CertInfo, error) {
+	cfg := config.Get()
+
+	secret, err := ics.k8s.GetSecret(cfg.IstioNamespace, secretName)
```

Comment:
Again, if the secret can be found in any namespace under any name, we can't do this - it would require the "read-secret" permission cluster-wide. And we do not want to grant that - especially if this is just for displaying secret information (it isn't like we need this permission for Kiali to "do its job" - it is just for displaying info to the user).

We must be very careful whenever we grant read-secret permission to Kiali - and we must be VERY VERY careful if we want to grant it cluster-wide - it opens up a potentially dangerous security hole. We don't want Kiali responsible for a security breach because it's service account was granted the ability to read secrets in the cluster.